src/deadline_worker_agent/aws/deadline/__init__.py

```diff
@@ -849,3 +856,34 @@ def record_sync_inputs_fail_telemetry_event(
         event_type="com.amazon.rum.deadline.worker_agent.sync_inputs_failure",
         event_details=details,
     )
+
+
+def record_success_fail_telemetry_event(**decorator_kwargs: Dict[str, Any]) -> Callable[..., F]:
+    """
+    Decorator to try catch a function. Sends a success / fail telemetry event.
+    :param ** Python variable arguments. See https://docs.python.org/3/glossary.html#term-parameter
+    """
+
+    def inner(function: F) -> F:
+        def wrapper(*args: List[Any], **kwargs: Dict[str, Any]) -> Any:
+            """
+            Wrapper to actually try-catch
+            :param * Python variable argument. See https://docs.python.org/3/glossary.html#term-parameter
+            :param ** Python variable argument. See https://docs.python.org/3/glossary.html#term-parameter
+            :
+            """
+            success: bool = True
+            try:
+                return function(*args, **kwargs)
+            except Exception as e:
+                success = False
+                raise e
+            finally:
+                _get_deadline_telemetry_client().record_event(
+                    event_type=f"com.amazon.rum.deadline.worker_agent.{function.__name__}",
```

Comment:
I liked how in another PR, you also supported explicitly adding the event name so code could be refactored without changing the telemetry. Might be useful here, e.g. to cleanup the leading "_" for one of the functions.

Reply:
Good point - Let me port over the code. The reason they are different is because the namespace here is different.

I wrote this one first, and then did the "event_name" change in the deadline cloud library and noticed also the `_` private method. easy add - will do this.

Reply:
Updated!